torch/csrc/distributed/rpc/tensorpipe_agent.cpp

```diff
@@ -572,12 +570,14 @@ void TensorPipeAgent::respond(std::shared_ptr<tensorpipe::Pipe>& pipe) {
                   << " is running request #" << messageId << " from "
                   << pipe->getRemoteName() << " in thread pool";
 
-          std::shared_ptr<FutureMessage> futureResponseMessage;
+          auto futureResponseMessage =
+              std::make_shared<JitFuture>(at::AnyClassType::get());
```

Comment:
Same as above: we're allocating an uncompleted JitFuture just to immediately replacing it (and thus deallocating it). It's probably better to leave the shared_ptr unset.